external-crates/move/crates/move-compiler/src/shared/ide.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{expansion::ast as E, parser::ast as P, shared::Name, naming::ast as N, typing::ast as T, ice, diagnostics::Diagnostics};
+
+use move_ir_types::location::Loc;
+
+use std::collections::BTreeMap;
+
+#[derive(Debug, Clone)]
+pub struct IDEInfo {
+    pub exp_info: BTreeMap<Loc, ExpEntry>
+}
+
+#[derive(Debug, Clone)]
+pub struct ExpEntry {
+    /// Location of the recorded info
+    pub loc: Loc,
```

Comment:
Is this a different location than the one used as key of the `exp_info` map (e.g., indicates a different, more precise location inside of the expression)?

Reply:
No, it's just duplicated here for sanity.